src/Voron/Impl/Transaction.cs

```diff
@@ -705,8 +705,8 @@ public void ForgetAbout(in long storageId)
 
             if (_cachedDecompressedBuffersByStorageId.Remove(storageId, out var t))
             {
-                Allocator.Release(ref t);
                 _lowLevelTransaction.DecompressedBufferBytes -= t.Length;
+                Allocator.Release(ref t);
```

Comment:
this fix should go to 5.4 as well

Reply:
https://github.com/ravendb/ravendb/pull/19627